### ISSUE TITLE
Width argument not taken into account

### ISSUE THREAD

**Issue by Kochise** [mermaid-js/mermaid-cli#73]
As mentioned here : https://github.com/mermaid-js/mermaid/issues/341?ts=4#issuecomment-725992429

npx mmdc --theme "neutral" -w "1000" -i state.mmd -o state.png
npx mmdc --theme "neutral" -w 2048 -i state.mmd -o state.png
npx mmdc --theme "forest" --width "2048" -i state.mmd -o state.png
npx mmdc --theme "neutral" -w "800" -i state.mmd -o state.png
npx mmdc --theme "neutral" -H "2048" -i state.mmd -o state.png
...

None of them change a thing, I'm still getting the same 202x346 pixels picture.

See attached picture below.

![state](https://user-images.githubusercontent.com/23077417/98931796-ca482f00-24de-11eb-953c-7d58c7c050eb.png)

I need a better resolution to import into a LaTeX documentation.

**Comment by MindaugasLaganeckas** [comment]
Could be related to #55.

**Comment by MindaugasLaganeckas** [comment]
@Kochise : the issue has been fixed. Please, reopen if you disagree.

**Comment by multimeric** [comment]
It certainly seems to be ignored for me. Here's my reproducible example, just using the example diagram from the mermaid website:
```
$ mmdc -i test.mmd -o test.svg -w 200
Generating single mermaid chart
$ mmdc --version
9.3.0
$ cat test.mmd
flowchart LR

A[Hard] -->|Text| B(Round)
B --> C{Decision}
C -->|One| D[Result 1]
C -->|Two| E[Result 2]
```

![image](https://user-images.githubusercontent.com/5019367/212581385-7404d00d-60a5-4387-9fea-a7d172675ab8.png)

**Comment by tildejustin** [comment]
having the same issue, `mmdc -i state.mmd -o state.png -t neutral --width 1920 --height 1080` with state.mmd of
```
stateDiagram-v2
%%{init: { "fontFamily": "Jetbrains Mono" } }%%

state "set target as current" as target_current
state "set target" as target
state IDLE {
    run_to_position --> run_to_position : target
}
state MANUAL {
    set_power --> set_power : gamepad input
}

[*] --> target_current
target_current --> IDLE
IDLE --> target : on any dpad input
note right of target
    up = high junction
    left & right = mid junction
    down = low junction
end note
target --> IDLE
IDLE --> MANUAL: on gamepad input w/o touching sensor
MANUAL --> target_current : stopped gamepad input
MANUAL --> target_current : up or down sensor pressed
```
always generates a `state.png` with a size of 986x587, whereas without width arguments it comes out as 784x468. This seems to be a cap on size, because 200x300 works just fine but 20000x30000 also gives 986x587.

**Comment by lufixSch** [comment]
@tildejustin I'm observing the same issue any news on that. Did you find a workaround/fix?

**Comment by tildejustin** [comment]
> @tildejustin I'm observing the same issue any news on that. Did you find a workaround/fix?

I don't think I did unfortunately, probably just upscaled the output somehow.

**Comment by wjkoh** [comment]
This problem still persists. Can we reopen it, @MindaugasLaganeckas ?

**Comment by NixImagery** [comment]
TL;DR: width `-w` is ignored or capped, but scale `-s` gets round it.

Using the same source and commands as @multimeric three years ago, I am finding this behaviour to be present in my current version of mmdc.

```{zsh}
% mmdc -i test.mmd -o test.png -w 1200
% exiftool test.png 
ExifTool Version Number         : 13.43
File Name                       : test.png
Directory                       : .
File Size                       : 13 kB
File Modification Date/Time     : 2026:02:04 10:47:18+00:00
File Access Date/Time           : 2026:02:04 10:47:19+00:00
File Inode Change Date/Time     : 2026:02:04 10:47:18+00:00
File Permissions                : -rw-r--r--
File Type                       : PNG
File Type Extension             : png
MIME Type                       : image/png
Image Width                     : 623
Image Height                    : 174
Bit Depth                       : 8
Color Type                      : RGB
Compression                     : Deflate/Inflate
Filter                          : Adaptive
Interlace                       : Noninterlaced
SRGB Rendering                  : Perceptual
Image Size                      : 623x174
Megapixels                      : 0.108

% mmdc --version
11.12.0

 % cat test.mmd 
flowchart LR

A[Hard] -->|Text| B(Round)
B --> C{Decision}
C -->|One| D[Result 1]
C -->|Two| E[Result 2]
```

A workaround seems to be to assume the output will be about 623 pixels but to scale it instead for a better resolution output:

```{zsh}
% mmdc -i test.mmd -o test.png -s 3   
Generating single mermaid chart

% exiftool test.png                
ExifTool Version Number         : 13.43
File Name                       : test.png
Directory                       : .
File Size                       : 40 kB
File Modification Date/Time     : 2026:02:04 10:59:00+00:00
File Access Date/Time           : 2026:02:04 10:59:02+00:00
File Inode Change Date/Time     : 2026:02:04 10:59:00+00:00
File Permissions                : -rw-r--r--
File Type                       : PNG
File Type Extension             : png
MIME Type                       : image/png
Image Width                     : 1869
Image Height                    : 522
Bit Depth                       : 8
Color Type                      : RGB
Compression                     : Deflate/Inflate
Filter                          : Adaptive
Interlace                       : Noninterlaced
SRGB Rendering                  : Perceptual
Image Size                      : 1869x522
Megapixels                      : 0.976
```
